### ISSUE TITLE
OpenStack modules don't work with cloud specified auth

### ISSUE THREAD

**Issue by Thingee** [ansible/ansible#19617]
_From @gunph1ld on October 13, 2016 13:21_

##### ISSUE TYPE
- Bug Report
##### COMPONENT NAME

os_keystone_service, os_user, etc..
##### ANSIBLE VERSION

```
ansible 2.2.0
  config file = 
  configured module search path = Default w/o overrides
```
##### CONFIGURATION

```
[defaults]
hostfile = hosts
host_key_checking = False
retry_files_enabled = False
#deprecation_warnings=False
#display_skipped_hosts=False

[paramiko_connection]
record_host_keys = False

[ssh_connection]
ssh_args = -o ControlMaster=auto -o ControlPersist=60s -o UserKnownHostsFile=/dev/null  -o IdentitiesOnly=yes
```
##### SUMMARY

I get an error with modules for OpenStack. `openstack` cli works well with the same parameters
##### STEPS TO REPRODUCE

```
- os_keystone_service:
    cloud: local_bypass
    state: present
    name: keystone
    service_type: identity
    description: 'OpenStack Identity'
```

```
[root@domain ~]# cat /etc/openstack/clouds.yaml 
clouds:
  local_bypass:
    auth_type: token_endpoint
    identity_api_version: 3
    auth:
      token: QWESDFwse423fs
      url: http://domain.com:35357/v3
```
##### EXPECTED RESULTS

Create a service suceffull
##### ACTUAL RESULTS

```
fatal: [domain.com]: FAILED! => {
    "changed": false, 
    "failed": true, 
    "invocation": {
        "module_args": {
            "api_timeout": null, 
            "auth": null, 
            "auth_type": null, 
            "availability_zone": null, 
            "cacert": null, 
            "cert": null, 
            "cloud": "local_bypass", 
            "description": "OpenStack Identity", 
            "enabled": true, 
            "endpoint_type": "public", 
            "key": null, 
            "name": "keystone", 
            "region_name": null, 
            "service_type": "identity", 
            "state": "present", 
            "timeout": 180, 
            "verify": true, 
            "wait": true
        }, 
        "module_name": "os_keystone_service"
    }, 
    "msg": "Problem with auth parameters"
}
```

but with `openstack-cli` it works well

```
# openstack --os-cloud local_bypass service create --name keystone --description "OpenStack Identity" identity
+-------------+----------------------------------+
| Field       | Value                            |
+-------------+----------------------------------+
| description | OpenStack Identity               |
| enabled     | True                             |
| id          | c70acc58b250421d8084c2a26f358e88 |
| name        | keystone                         |
| type        | identity                         |
+-------------+----------------------------------+
```

_Copied from original issue: ansible/ansible-modules-core#5250_

**Comment by Thingee** [comment]
_From @gunph1ld on October 13, 2016 13:40_

The same behavior if use credentials directly in the module:

```
- os_keystone_service:
    auth_type: token_endpoint
    auth:
      token: QWESDFwse423fs
      url: http://domain.com:35357/v3
    state: present
    name: keystone
    service_type: identity
    description: 'OpenStack Identity'
```

```
fatal: [domain.com]: FAILED! => {
    "changed": false, 
    "failed": true, 
    "invocation": {
        "module_args": {
            "api_timeout": null, 
            "auth": {
                "token": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
                "url": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER"
            }, 
            "auth_type": "token_endpoint", 
            "availability_zone": null, 
            "cacert": null, 
            "cert": null, 
            "cloud": null, 
            "description": "OpenStack Identity", 
            "enabled": true, 
            "endpoint_type": "public", 
            "key": null, 
            "name": "keystone", 
            "region_name": null, 
            "service_type": "identity", 
            "state": "present", 
            "timeout": 180, 
            "verify": true, 
            "wait": true
        }, 
        "module_name": "os_keystone_service"
    }, 
    "msg": "Problem with auth parameters"
}
```

**Comment by Thingee** [comment]
_From @ansibot on October 13, 2016 13:43_

@emonty, @shrews, @juliakreger, @j2sol, @rcarrillocruz, ping. This issue is waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by Thingee** [comment]
_From @emonty on October 24, 2016 3:44_

Made a patch here:
https://review.openstack.org/390148 

token_endpoint isn't actually a valid auth plugin - it is a python-openstackclient specific thing. However, it's still a pretty crappy experience for the user to hit against that - so I think making os-client-config do what you meant there is fair.

**Comment by Thingee** [comment]
_From @ansibot on October 29, 2016 8:46_

@emonty, @shrews, @juliakreger, @j2sol, @rcarrillocruz, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by Thingee** [comment]
_From @ansibot on November 13, 2016 9:53_

@emonty, @shrews, @juliakreger, @j2sol, @rcarrillocruz, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by Thingee** [comment]
_From @ansibot on November 29, 2016 9:34_

@emonty, @shrews, @juliakreger, @j2sol, @rcarrillocruz, @thingee, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by Thingee** [comment]
_From @ansibot on December 7, 2016 19:18_

This repository has been locked. All new issues and pull requests should be filed in https://github.com/ansible/ansible

Please read through the <a href="http://docs.ansible.com/ansible/dev_guide/repomerge.html#move-issues-and-prs-to-new-repo">repomerge</a> page in the dev guide. The guide contains links to tools which automatically move your issue or pull request to the ansible/ansible repo.

**Comment by ktibi** [comment]
I have same issue with dynamic inventory. 

Works with env OS_* or if clouds.yml is in my current directory but no works if the file is in another directory like .config/openstack/clouds.yml or /etc/ansible/.....

```
Couldn't construct compute object
Traceback (most recent call last):
  File "/root/.virtualenvs/galaxy/lib/python2.7/site-packages/shade/openstackcloud.py", line 354, in _get_client 
  **kwargs)
  File "/root/.virtualenvs/galaxy/lib/python2.7/site-packages/os_client_config/cloud_config.py", line 322, in get_legacy_client
    endpoint = self.get_session_endpoint(service_key)
  File "/root/.virtualenvs/galaxy/lib/python2.7/site-packages/os_client_config/cloud_config.py", line 256, in get_session_endpoint
    session = self.get_session()
  File "/root/.virtualenvs/galaxy/lib/python2.7/site-packages/os_client_config/cloud_config.py", line 199, in get_session
    "Problem with auth parameters")
OpenStackConfigException: Problem with auth parameters
Manager OpenstackCloud: ran task ServerList in 0.00397419929504s
Error fetching server list on OpenstackCloud::
```

**Comment by ansibot** [comment]
cc @SamYaple
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
cc @dagnello
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by timeu** [comment]
ran into the same issue. Is there any update ?

**Comment by ansibot** [comment]
cc @omgjlk
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
cc @mnaser
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by mnaser** [comment]
@ktibi can you please share the task you're trying to run, as well as try using the openstackclient and let me know if you have issues?

**Comment by mnaser** [comment]
needs_info

**Comment by ansibot** [comment]
cc @cloudnull @evrardjp @odyssey4me
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@Thingee This issue is waiting for your response. Please respond or the issue will be closed.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@Thingee You have not responded to information requests in this issue so we will assume it no longer affects you. If you are still interested in this, please create a new issue with the requested information.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)
